addon/@ember/test-waiters/waiter-manager.ts

```diff
@@ -1,39 +1,69 @@
+import { assert } from '@ember/debug';
 import { PendingWaiterState, Waiter, WaiterName } from './types';
 
 import Ember from 'ember';
 import { registerWaiter } from '@ember/test';
 
-type Indexable = Record<any, unknown>;
+assert(
+  `Expected the 'Symbol' global to be available in this environment. Environments without support for 'Symbol' are not supported by '@ember/test-waiters'. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol`,
+  typeof Symbol !== 'undefined'
+);
+
+// All copies of `@ember/test-waiters must have this same key/symbol.
+// Once this `@ember/test-waiters` is released, folks will want to use 'overrides'
+// to pin it across their whole dep graph.
+//
+// In the v1 addon version of this library, the build time highlander code will ensure
+// that there is only one copy,
+// And with the v2 addon version of this library, that is where pinning is important,
+// because either:
+// - highlander code will run, and boot out the v2 addon copy
+// - v2 addon has precedence, and the min-version of test-waiters throughout the
+//   dep graph should not preceed the version which this symbol was introduced.
+const PRIVATE_GLOBAL_DATA_KEY = Symbol.for(`@ember/test-waiters' WAITERS`);
```

Comment:
So I am still confused. The existing code in this file seems to have already made the effort to avoid the problem you described, as indicated by the comment below. Because it effectively uses a global variable, if inadvertently the build code fails to dedup the copies, the multiple copies will still share the same map. This code now changes the symbol name and the structure of the variable, thus defeating the previous attempt and breaking compatibility. Why is that a good idea?

Reply:
no I'm is wrong, the data structures are incompatible. I let you sway me too easily, thinking I've done soething wrong!

the _data stored on the symbol on glabalThis_ is changing:
- instead of a Map its an object with Map and a Set
- this change is because the Set is a logging system to prevent log spew.

So instead, to keep the data structures compatible, I would need to revert this change:
```js
globalThis: {
  [Symbol<PRIVATE_GLOBAL_DATA_KEY>]?: {
    WAITER_NAMES?: Set<string> | undefined;
    WAITERS?: Map<WaiterName, Waiter> | undefined;
  }
}
```
and instead do:
```js
globalThis: {
  [Symbol<TEST_WAITERS>]?: Map<WaiterName, Waiter> | undefined;
  [Symbol<TEST_WAITER_NAMES>]?: Set<string> | undefined;
}
```

Which does add the things on globalThis, but is probably a safer change